locations/spiders/coopfood.py

```diff
@@ -0,0 +1,49 @@
+import json
+import re
+import scrapy
+from locations.items import GeojsonPointItem
+class CoopFoodSpider(scrapy.Spider):
+    name = "coopfood"
+    allowed_domains = ["api.coop.co.uk"]
+    download_delay = 0.5
+    state = True
+    def start_requests(self):
+        for page in range(1000):
+            if(self.state==False):
+                break
+            else:
+                url = 'https://api.coop.co.uk/locationservices/finder/food/?location=54.9966124%2C-7.308574799999974&distance=30000000000&always_one=true&format=json&page='+str(page)
+                yield scrapy.Request(
+                    url=url,
+                    callback=self.parse
+                )
+
+    def parse(self, response):
+      try:
+        data = json.loads(response.body_as_unicode())
+        if(data['next']==None):
+            self.state = False
```

Comment:
Rather than using this `state` variable, you can start with the first page in the `start_requests` list and then in `parse` yield another `scrapy.Request()` for the next page after yielding the items. Then when there's no more results you can stop yielding requests. That would clean this up a bit.